src/matrix_pascal.c

```
@@ -0,0 +1,29 @@
+#include <stdio.h>
+#include <stdlib.h>
+
+#include "pascal_functions.h"
+
+int main() {
+    int matrix_size;
+    if (scanf("%d", &matrix_size) == 1 && matrix_size > 0 && matrix_size <= 10) {
```

Так, надо хоть к чему-то придраться )))
Я бы сразу тут вышел при некорректном вводе, до else далеко листать.

---

программа листать не будет, сразу в else перейдет)
Писать код от обратного? сначала ошибки а в else саму суть?))

---

В одной умной книге подцепил правило "Fail early" - при любых ошибках которые не подразумевают корректной работы программы вываливаться сразу.

---

Т.е. тогда else не будет. if всё плохо - сообщение в stderror и exit_failure. А ниже просто продолжение кода. И вложенность уменьшится и сразу видна реакция на некорректный ввод.